runtime/common/src/assigned_slots/mod.rs

```diff
@@ -377,15 +407,48 @@ pub mod pallet {
 					// Treat failed downgrade as warning .. slot lease has been cleared,
 					// so the parachain will be downgraded anyway by the slots pallet
 					// at the end of the lease period .
-					log::warn!(target: "assigned_slots",
+					log::warn!(
+						target: LOG_TARGET,
 						"Failed to downgrade parachain {:?} at period {:?}: {:?}",
-						id, Self::current_lease_period_index(), err
+						id,
+						Self::current_lease_period_index(),
+						err
 					);
 				}
 			}
 
 			Ok(())
 		}
+
+		// TODO: Benchmark this
+		/// Assign a max permanent slot number.
+		#[pallet::call_index(3)]
+		#[pallet::weight(((MAXIMUM_BLOCK_WEIGHT / 10) as Weight, DispatchClass::Operational))]
+		pub fn set_max_permanent_slots(origin: OriginFor<T>, slots: u32) -> DispatchResult {
+			ensure_root(origin)?;
+
+			<MaxPermanentSlots<T>>::put(slots);
+
+			// Emit an event.
+			Self::deposit_event(Event::<T>::MaxPermanentSlotsAssigned { slots });
+			// Return a successful DispatchResult
+			Ok(())
+		}
+
+		// TODO: Benchmark this
```

Comment:
Why is nothing in this pallet benchmarked :see_no_evil:

Reply:
Added!
Not sure about how to proceed with the weights, Do I have to generate the weights file using my machine? 
Or do something like the auctions: 
https://github.com/paritytech/polkadot/blob/master/runtime/common/src/auctions.rs#L50
https://github.com/paritytech/polkadot/blob/master/runtime/common/src/auctions.rs#L283